app/controllers/api/admin/articles_controller.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class Api::Admin::ArticlesController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    if current_user.role == 'editor'
```

Comment:
you can extract this conditional to a `:before_action`

Reply:
Not necessarily as we will later want to have journalist being able to view their own unpublished articles (which means the show action cannot have this). So currently, yes, but the refactoring would be unnecessary as it would need to be undone later.

Reply:
**later**

You will need to check with every request to this controller that the user is an editor, journalist, or admin. Otherwise, they are unauthorized. There is no reason to not extract checking the user role to a before action.

Reply:
But editor, journalist and admin have different checks.
So sure, I could change the before_action to do a check if the user has role >= journalist, but I will for most cases need to separate check anyway.
Why should I refactor if not for easier later? It is perfectly clear what the check is doing right now.
If you insist that it is necessary, please do, and I will change it, but I still think it will not make it better, neither in the long run nor immediately.
At a later point it makes sense to extract the unauthorized message render.
If anything it would make more sense to REMOVE the before action currently in and just have custom checks.

Reply:
Authenticate user is to check that the user is  logged in

The before action i’m asking you to add is to check that the user trying to access it has the right role to access it

Instead if having a conditional in every possible action u will have in that controller to check if the user has the role of “registered_user” or “subscriber” and therefore unauthorized. I want u to add it to a before action. 

You will still check the role if journalist, editor and admin. But then u can skip having conditionals for non staff roles. 

Either u extract it to a before action or you start using pundit. It will no be merged in its current state.

Reply:
Added big and clunky check for role. Added the role subscriber to the role, as we are checking for it anyway, even though it has no purpose yet.
Other plan would be to just do a check current_user.role >= 'editor', in the action, it will do all the work and none of the hassle. But team voted not to so... clunky code.